hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/token/delegation/AbstractDelegationTokenSecretManager.java

```diff
@@ -67,9 +67,9 @@
 
 @InterfaceAudience.Public
 @InterfaceStability.Evolving
-public abstract 
-class AbstractDelegationTokenSecretManager<TokenIdent 
-extends AbstractDelegationTokenIdentifier> 
+public abstract
```

Comment:
I am a little worried that changes in this class may affect many sub classes
<img width="1672" alt="image" src="https://user-images.githubusercontent.com/55643692/222172441-dee6dd5f-2d6b-4a8a-a556-c2b727a43ea4.png">

Reply:
In general, this is a fairly fundamental class in Hadoop in general.
I would propose to do a separate JIRA to clean this class up.

Reply:
I don't think any of these changes will affect any other functionality in any sub class - I verified the same . 

And the changes are either very generic like encapsulation, exception handling, concurrency handling, etc - let me know if this still warrants a seperate JIRA.